core/src/main/java/org/infinispan/interceptors/impl/AsyncInterceptorChainImpl.java

```diff
@@ -230,6 +231,7 @@ public void appendInterceptor(AsyncInterceptor ci, boolean isCustom) {
 
    @Override
    public CompletableFuture<Object> invokeAsync(InvocationContext ctx, VisitableCommand command) {
+      Stages.pushInvocationIdNDC(command, trace);
```

Comment:
Ok, I removed the logging changes :)

I added the NDC because I was trying to figure out a timeout error in a query test, and following the handling of a write command that triggers 10s of writes in the index cache is pretty painful. In the end it turned out to be just trace logging slowing things down, but I still want to add this in another PR.